src/libraries/Microsoft.PowerFx.Core/Public/Config/NameSymbol.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Used between Symbol Table resolver and IR.
+    /// NameInfo data to associate a symbol at bind time with a runtime config at runtime. 
+    /// Object provided by resolver data in <see cref="NameLookupInfo.Data"/>.
+    /// IR then recognizes this and will look these values up via ReadOnlySymbolValues.TryGetValue.
+    /// </summary>
+    [DebuggerDisplay("{Name} ({Parent.DebugName}:{SlotIndex})")]
+    internal class NameSymbol : ISymbolSlot
+    {
+        public NameSymbol(string name, bool mutable)
+        {
+            Name = name;
+            IsMutable = mutable;
+        }
+
+        public string Name { get; private set; }
+
+        /// <summary>
+        /// Can this variable be passed to Set(). If not, then this is a binding failure. 
+        /// </summary>
+        public bool IsMutable { get; private set; }
+
+        public ReadOnlySymbolTable Parent { get; set; }
+
+        public int SlotIndex { get; set; } = -1;
+
+        public void Dispose()
```

Comment:
>Dispose [](http://example.com/codeflow?start=20&length=7)

We should use a different name to avoid confusions with IDisposable

Reply:
DisposeSlot()